FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.HealthCheck/Checkers/AzureBlobStorageHealthChecker.cs

```diff
@@ -0,0 +1,85 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Synapse.Common.Configurations;
+using Microsoft.Health.Fhir.Synapse.DataWriter.Azure;
+using Microsoft.Health.Fhir.Synapse.HealthCheck.Models;
+
+namespace Microsoft.Health.Fhir.Synapse.HealthCheck.Checkers
+{
+    public class AzureBlobStorageHealthChecker : BaseHealthChecker
+    {
+        public const string HealthCheckBlobPrefix = ".healthcheck";
+        private readonly IAzureBlobContainerClient _blobContainerClient;
+
+        public AzureBlobStorageHealthChecker(
+            IAzureBlobContainerClientFactory azureBlobContainerClientFactory,
+            IOptions<JobConfiguration> jobConfiguration,
+            IOptions<DataLakeStoreConfiguration> storeConfiguration,
+            ILogger<AzureBlobStorageHealthChecker> logger)
+            : base(HealthCheckTypes.AzureBlobStorageCanReadWrite, false, logger)
+        {
+            EnsureArg.IsNotNull(azureBlobContainerClientFactory, nameof(azureBlobContainerClientFactory));
+            EnsureArg.IsNotNull(jobConfiguration, nameof(jobConfiguration));
+            EnsureArg.IsNotNull(storeConfiguration, nameof(storeConfiguration));
+
+            _blobContainerClient = azureBlobContainerClientFactory.Create(storeConfiguration.Value.StorageUrl, jobConfiguration.Value.ContainerName);
+        }
+
+        protected override async Task<HealthCheckResult> PerformHealthCheckImplAsync(CancellationToken cancellationToken)
+        {
+            var healthCheckResult = new HealthCheckResult(HealthCheckTypes.AzureBlobStorageCanReadWrite, false);
+            var blobPath = $"{HealthCheckBlobPrefix}";
+
+            try
+            {
+                // Ensure we can write to the storage account
+                byte[] bytes = Encoding.UTF8.GetBytes(HealthCheckBlobPrefix);
```

Comment:
Shall we use another variable for the blob value? Reusing the prefix might be confusing in the future